pkgs/development/web/bun/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchurl, autoPatchelfHook, unzip, openssl, lib }:
+let
+  version = "0.1.1";
+in
+stdenv.mkDerivation {
+  pname = "bun";
+  version = version;
+  src = fetchurl {
+    url = "https://github.com/Jarred-Sumner/bun-releases-for-updater/releases/download/bun-v${version}/bun-linux-x64.zip";
+    sha256 = "sha256-UcBNcWJ7+ZhoUn9qADl6uovAE6kOHvGgIICk6T0pHA0=";
```

Comment:
`bun` supports both `linux` and `darwin`, but here only uses linux release output.

How about build `bun` from the source?

Reply:
Yeah, I was trying to avoid having to do that but it may be necessary. The problem is that due to the nature of bun it's a bit of a pain to build from source since it uses forks of zig and the entirety of WebKit and JavaScriptCore which themselves have their own finicky platform specific build jazz etc. But I suppose it may be necessary? What do you think? I'm willing to figure it out sometime this week if need be.

Reply:
I found the zig is currently marked as broken on darwin (#86299), so may not be able to patched in nix without upstream fix.

And bun uses patched `webkit` on linux during the build by zig, and produces cross-platform outputs. I'm not sure the `webkit` can be built on darwin with nix.

I'm just one of the nixpkgs contributors, but I'd probably use tarballs and zip files from bun github releases. (since the build is quite difficult and looks hacky) In this case there should be different url & sha256 per platform, and exact `meta.platforms` value is required.

Reply:
I hope that someone who knows zig and nix well come.

Reply:
I'll go with the tarballs for now. I've hacked on zig with nixos a bunch a while back and the story for it isn't great yet. I have no Darwin machines so I'll just have it grab the right tarball and trust that it will work.